File: front-end-app/app/src/main/java/com/dev/hirelink/enums/JobApplicationState.kt
package com.dev.hirelink.enums

import com.dev.hirelink.R

enum class JobApplicationState(val colorResourceId: Int, val stringResourceId: Int) {
    PENDING(R.color.indigo_main, R.string.job_application_pending_state),
    IN_PROGRESS(R.color.amber_600, R.string.job_application_in_progress_state),
    ACCEPTED(R.color.state_green, R.string.job_application_accepted_state),
    REFUSED(R.color.red_A700, R.string.job_application_refused_state)
}

File: backend-app/src/Enum/JobApplicationState.php
<?php

namespace App\Enum;

enum JobApplicationState
{
    case PENDING;
    case IN_PROGRESS;
    case ACCEPTED;
    case REFUSED;

    public static function fromCode(string $code): JobApplicationState
    {
        return match(true) {
            $code === ACCEPTED_CODE => JobApplicationState::ACCEPTED,
            $code === REFUSED_CODE => JobApplicationState::REFUSED,
            $code === IN_PROGRESS_CODE => JobApplicationState::IN_PROGRESS
        };
    }
}